### ISSUE TITLE
Issue with file path with Cypress component testing

### ISSUE THREAD

**Issue by Sett17** [BBE78/cypress-sonarqube-reporter#46]
**Describe the bug**
Running this reporter with Cypress CT results in this:

```
could not find spec filename from title: redacted or from 'suite.invocationDetails'
Error: could not find spec filename from title: redacted or from 'suite.invocationDetails'
    at throwError (/home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/ReporterUtils.js:25:11)
    at extractSpecFromSuite (/home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/ReporterUtils.js:51:9)
    at SonarQubeCypressReporter.traverseSuite (/home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:71:33)
    at /home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:82:18
    at Array.forEach (<anonymous>)
    at SonarQubeCypressReporter.traverseSuite (/home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:81:22)
    at SonarQubeCypressReporter.onDone (/home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:58:14)
    at L.<anonymous> (/home/redacted/redacted/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.15.0/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:45:18)
    at Object.onceWrapper (node:events:642:26)
    at L.emit (node:events:539:35)
    at E.emit (<embedded>:4582:16654)
    at Object.onMocha (<embedded>:4629:389133)
    at p.<anonymous> (<embedded>:4629:30585)
    at p.emit (node:events:527:28)
    at p.emitUntyped (<embedded>:4340:84346)
    at <embedded>:4340:91863
    at process.processTicksAndRejections (node:internal/process/task_queues:78:11)
```

From what I've seen, there is no useful filepath anywhere in the suite given to this function (or the function before this) except in the `parent`.
Locally, this change works, as expected:
```diff
--- a/src/ReporterUtils.js
+++ b/src/ReporterUtils.js
@@ -27,6 +27,7 @@ const throwError = (message) => {

 /**
  * Extract the spec file path from the suite.
+ * May ignore the options.useAbsoluteSpecPath if the requested is not available.
  * Raise an error if the spec could not be extracted.
  *
  * @param {object} suite the Mocha suite
@@ -41,8 +42,10 @@ const extractSpecFromSuite = (suite, options) => {
     if (index > -1) {
         spec = JSON.parse(title.substring(index + tag.length, title.lastIndexOf(']')));
         spec = options.useAbsoluteSpecPath ? spec.absolute : spec.relative;
-    } else if (suite.invocationDetails) {
+    } else if (suite.invocationDetails && (suite.invocationDetails.absoluteFile || suite.invocationDetails.relativeFile)) {
         spec = options.useAbsoluteSpecPath ? suite.invocationDetails.absoluteFile : suite.invocationDetails.relativeFile;
+    } else if (suite.parent && suite.parent.file) {
+        spec = suite.parent.file;
     } else {
         throwError(`could not find spec filename from title: ${title} or from 'suite.invocationDetails'`);
     }
```

I am not sure if this is really a fix (if so, I'd happily open a PR) or just a workaround.

**Comment by TomaszG** [comment]
+1

I have a similar issue with component testing:
```
Cannot read properties of undefined (reading 'replace')
TypeError: Cannot read properties of undefined (reading 'replace')
    at extractSpecFromSuite (/Users/<redacted>>/Projects/<redacted>/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.17.4/node_modules/cypress-sonarqube-reporter/src/ReporterUtils.js:49:17)
    at SonarQubeCypressReporter.traverseSuite (/Users/<redacted>>/Projects/<redacted>/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.17.4/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:72:33)
    at /Users/<redacted>>/Projects/<redacted>/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.17.4/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:83:18
    at Array.forEach (<anonymous>)
    at SonarQubeCypressReporter.traverseSuite (/Users/<redacted>>/Projects/<redacted>/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.17.4/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:82:22)
    at SonarQubeCypressReporter.onDone (/Users/<redacted>>/Projects/<redacted>/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.17.4/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:58:14)
    at R.<anonymous> (/Users/<redacted>>/Projects/<redacted>/node_modules/.pnpm/cypress-sonarqube-reporter@1.12.0_cypress@12.17.4/node_modules/cypress-sonarqube-reporter/src/SonarQubeCypressReporter.js:45:18)
    at Object.onceWrapper (node:events:642:26)
    at R.emit (node:events:539:35)
    at S.emit (<embedded>:4514:16654)
    at Object.onMocha (<embedded>:4561:139329)
    at p.<anonymous> (<embedded>:4561:30585)
    at p.emit (node:events:527:28)
    at p.emitUntyped (<embedded>:4458:84346)
    at <embedded>:4458:91863
    at process.processTicksAndRejections (node:internal/process/task_queues:78:11)
 ELIFECYCLE  Command failed with exit code 1.
```

The fix above proposed by OP seems to help. Can I ask any maintainer to release it?

**Comment by Sett17** [comment]
@TomaszG You could bump the already open PR #47

**Comment by Sett17** [comment]
If anyone needs this fix before it is merged, it is possible to use the PR branch as the package.
```json
"cypress-sonarqube-reporter": "github:Sett17/cypress-sonarqube-reporter",
```

This should, of course, be only used for testing or if desperately needed, and changed once the fix is merged

**Comment by BBE78** [comment]
Hi @Sett17, @TomaszG I'm pleased to announce that @Sett17 fix is now available in v1.13.0
Good job 👌
